base/task.jl

```diff
@@ -420,7 +423,12 @@ function _wait_multiple(waiting_tasks; all=false, failfast=false)
     if nremaining == 0
         return tasks, Task[]
     else
-        return tasks[done_mask], tasks[.~done_mask]
+        remaining_tasks = tasks[.~done_mask]
+        for t in remaining_tasks
+            waiter = pop!(waiter_tasks, t)
+            @lock t.donenotify Base.list_deletefirst!((t.donenotify::ThreadSynchronizer).waitq, waiter)
+        end
+        return tasks[done_mask], remaining_tasks
```

Comment:
@vtjnash I've added this clean-up code to delete waiter tasks from the `waitq` of each remaining task.
I introduced the `waiter_tasks` dictionary to obtain the corresponding waiter task `waiter` from the target task `t`.